dvc/command/stage.py

```diff
@@ -194,3 +292,47 @@ def add_parser(subparsers, parent_parser):
     )
     _add_common_args(stage_add_parser)
     stage_add_parser.set_defaults(func=CmdStageAdd)
+
+    STAGE_LIST_HELP = "List stages."
+    stage_list_parser = stage_subparsers.add_parser(
+        "list",
+        parents=[parent_parser],
+        description=append_doc_link(STAGE_LIST_HELP, "stage/list"),
+        help=STAGE_LIST_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    stage_list_parser.add_argument(
+        "targets",
+        nargs="*",
+        default=["dvc.yaml"],
+        help=(
+            "Show stages from a dvc.yaml/.dvc file or a directory. "
+            "'dvc.yaml' by default"
+        ),
+    )
+    stage_list_parser.add_argument(
+        "--all",
+        action="store_true",
+        default=False,
+        help="List all of the stages in the repo.",
```

Comment:
``"List all of the stages in the repo regardless of `targets`."`` (if I'm understanding this correctly)

Reply:
I find "regardless of `targets`" to be redundant due to the "all of the stages" there.

Reply:
It's good to be redundant sometimes. We need to be extra clear. Users are not machines.

Reply:
> Users are not machines. 

By this I mean that we often describe things correctly but people still get confused. All the commands an options are not obvious at all and as the developers we are biased because we already deeply understand them.